platform/qt/src/qmapboxgl.cpp

```diff
@@ -785,6 +894,21 @@ void QMapboxGL::setLayoutProperty(const QString& layer_, const QString& property
     }
 }
 
+/*!
+    Sets a paint \a property \a value to an existing \a layer_. The \a property string can be any
+    as defined by \l {https://www.mapbox.com/mapbox-gl-style-spec/} {the vector tiles specification}
+    for paint properties.
+
+    The argument \a klass_ is deprecated and is used for defining the style class for the paint
```

Comment:
My point was that you’d have two methods:

```cpp
void setPaintProperty(const QString& layer, const QString& property, const QVariant& value);
//! \deprecated
void setPaintProperty(const QString& layer, const QString& property, const QVariant& value, const QString& klass) __attribute__((deprecated));
```

This works in C++03; is there a Qt-specific reason why we can’t overload the method like this?